### ISSUE TITLE
"document" is not available during server side rendering.

### ISSUE THREAD

**Issue by hristov296** [thednp/kute.js#99]
I tried searching for this, but couldn't find relevant issues.
I am using kute to make animations in a gatsby generated site. When I do the gatsby `build` command, I get this error:

`
  17 |   var g = typeof global !== 'undefined' ? global : window, time = g.performance,
> 18 |     body = document.body, tweens = [], tick = null, // tick must be null!!
     | ^
  19 | 
  20 |     // strings
  21 |     length = 'length',

  WebpackError: ReferenceError: document is not defined
`
and a link to this page for more info: [https://gatsby.dev/debug-html](url)

Basically, gatsby does a server-side render, during which, `window` is not defined. I was able to fix this by adding `if (typeof window === 'undefined') { return; }` line before the line with `var g` (17) definition. This is for all kute modules, which I used. Do you think this should be included by default?

***I was also able to fix it, following their guide, by adding a piece of code, which excludes kute modules during ssr.***

**Comment by thednp** [comment]
@hristov296 thanks for the input. Please stand by and prepare for the KUTE.js v2.0 script coming soon to the MASTER branch, I will probably need your help there to solve these server issues.

Cheers.

**Comment by hristov296** [comment]
That's awesome, I'm waiting for it. When to expect it, estimated?

**Comment by thednp** [comment]
Hard to say, I'm working on documentation, perhaps 70-80% done. But the script is AWESOME both quality and performance.

**Comment by thednp** [comment]
You can help me by telling me what is the NAME of the global variable in the server?

**Comment by hristov296** [comment]
Do you mean like `window` in the browser? I don't think there is such. I am not too competent about these stuff

**Comment by thednp** [comment]
What if you define an empty object `{}` as global we need for that `g` var?

**Comment by hristov296** [comment]
The problem is on line 18
`    body = document.body, tweens = [], tick = null, // tick must be null!!`
when it sees `document`, it throws error `document is not defined`. During server side rendering there is `window` or `document`. I think gatsby(and any building tool like it) can just skip the module, because it doesn't do anything on the server side. I was able to do this by exiting early from the module, with a check for `typeof window === 'undefined'`.

`if (typeof window === 'undefined') { return; }`

![Screenshot_4](https://user-images.githubusercontent.com/18082688/75149440-a81d3f80-570a-11ea-97f6-432bf1a0a540.jpg)

If'd you like, you can give me some code and I can test it

**Comment by thednp** [comment]
OK, tell me how this works

var g = typeof global !== 'undefined' ? global : self||window||{}

**Comment by hristov296** [comment]
still throws the error 
![image](https://user-images.githubusercontent.com/18082688/75152760-760fdb80-5712-11ea-8cd8-54f15ebb5e77.png)

**Comment by thednp** [comment]
But no longer complain about `g`.

**Comment by hristov296** [comment]
Sorry, I might have worded it poorly. My last screenshot was the error all along.

**Comment by thednp** [comment]
Well then, why don't you wait a couple of days, you gonna build KUTE.js your way with rollup, OK?

**Comment by hristov296** [comment]
Yeah that is fine

**Comment by thednp** [comment]
To my understanding, you're using webpack for custom app.js builds?

**Comment by hristov296** [comment]
Yes it's webpack, but it's not custom. It's setup by [gatsby](https://github.com/gatsbyjs/gatsby). Specifically, their `npm run build` script

**Comment by thednp** [comment]
https://thednp.github.io/kute.js/
